src/networking.c

```diff
@@ -215,14 +215,31 @@ client *createClient(connection *conn) {
     return c;
 }
 
+void installClientWriteHandler(client *c) {
+    int ae_barrier = 0;
+    /* For the fsync=always policy, we want that a given FD is never
+     * served for reading and writing in the same event loop iteration,
+     * so that in the middle of receiving the query, and serving it
+     * to the client, we'll call beforeSleep() that will do the
+     * actual fsync of AOF to disk. the write barrier ensures that. */
+    if (server.aof_state == AOF_ON &&
```

Comment:
with #10616 changes, should we check AOF_WAIT_REWRITE in here? fsync the new INCR AOF file
```c
    /* Append to the AOF buffer. This will be flushed on disk just before
     * of re-entering the event loop, so before the client will get a
     * positive reply about the operation performed. */
    if (server.aof_state == AOF_ON ||
        (server.aof_state == AOF_WAIT_REWRITE && server.child_type == CHILD_TYPE_AOF))
    {
        server.aof_buf = sdscatlen(server.aof_buf, buf, sdslen(buf));
    }
```

Reply:
I don't think so.

First, this barrier thing (improved in this PR), is not really relevant in the `AOF_WAIT_REWRITE` state, since the data in redis it not yet persisted (no base file), so there's no point in adding guarantees that the incremental one is persisted before replying.

Secondly, note that #10616 didn't change anything in that regard. the topic of that PR was not to add the new (temporary) incremental file to the manifest before the base file (rewrite) is ready.

let me know if i'm missing anything.

Reply:
yes, after a re-thinking (i didn't go deep in that PR), there is no point doing it in the new temp INCR AOF, i missed it in the first place. thanks for the clarification